core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -1177,3 +1177,25 @@ class ZKCheckedEphemeral(path: String,
     }
   }
 }
+
+// Case classes for JSON deserialization
+
+/**
+  * Deserialized representation of a partition assignment.
+  *
+  * An assignment consists of a `version` and a list of `partitions`, which represent the assignment
+  * of topic-partitions to brokers.
+  */
```

Comment:
I think we should not document these classes as being for deserialization only. We should also use them for serialization so that we detect problems at that stage too.

Also, we can move these classes to `ZkData`.